### PR TITLE
Problem: archive node takes a long time to startup

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #666
Solution:
- update dependency minor versions
- update to golang 1.18 because some dependencies require that.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Improvements
+
+- [#]() Update iavl to 0.19.1 to improve node startup performance.
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
- [#667](https://github.com/crypto-org-chain/cronos/pull/667) Update iavl to 0.19.1 to improve node startup performance.
```

File: flake.lock [hunk 2]
```diff
@@ -56,16 +56,16 @@
...
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "269d35f70441a61bf1bd25350449a3518bfe7f7e",
+        "rev": "5342fc6fb59d0595d26883c3cadff16ce58e44f3",
         "type": "github"
       },
       "original": {
         "owner": "NixOS",
-        "ref": "release-21.11",
+        "ref": "master",
```
Comment (JayT106) [inline flake.lock:68]: why changes the dependent from the release version?

Comment (yihuang) [reply flake.lock:68]: It's required for golang 1.18 on macOS, it has a Mac sdk update.

**Review by thomas-nguy — APPROVED** [review]
lgtm

it is not a breaking change right? I there a documentation explaining what the optimization do? :)

**Comment by yihuang** [comment]
> it is not a breaking change right?

It should be not breaking since its patch version number bumps, but we'll be cautious nonetheless to gradually try on some full nodes first, and there's no need to use it on validator nodes at all, since validators use pruning and don't suffer from slow startup too much.

> Is there a documentation explaining what the optimization do? :)

the pr: https://github.com/cosmos/iavl/pull/468
the spec: https://hackmd.io/cMctM2NsRlCIojnW3-mo5Q

File: nix/scripts.nix [hunk 3]
```diff
@@ -6,12 +6,12 @@
...
+    export PATH=${pkgs.test-env}/bin:${chainmain}/bin:$PATH
     ${../scripts/start-chainmain} ${config.chainmain-config} ${config.dotenv} $@
   '';
   start-cronos = pkgs.writeShellScriptBin "start-cronos" ''
     # rely on environment to provide cronosd
-    export PATH=${pkgs.pystarport}/bin:$PATH
+    export PATH=${pkgs.test-env}/bin:$PATH
     ${../scripts/start-cronos} ${config.cronos-config} ${config.dotenv} $@
   '';
   start-geth = pkgs.writeShellScriptBin "start-geth" ''
```
Comment (mmsqe) [inline nix/scripts.nix:17]: start-geth might also need `test-env`

Comment (yihuang) [reply nix/scripts.nix:17]: yeah, indeed, strange that it works before.

**Pull request merged**